subsys/bootloader/bl_storage/bl_storage.c

```diff
@@ -92,6 +94,17 @@ int public_key_data_read(u32_t key_idx, u8_t *p_buf, size_t buf_size)
 		return -EFAULT;
 	}
 
+	for (u32_t n = 0; n < num_public_keys_read(); n++) {
```

Comment:
Is it needed to perform this for all keys, every time you call this function?

Reply:
I wanted this to fail loudly and early. I could put it into a separate function and call that from bl_validation once on each boot.

Reply:
Done